features/steps/thens/attributes.py

```diff
@@ -69,12 +69,14 @@ def step_impl(context, entity, other_entity):
 @then('The type of attribute {attribute} must be {expected_entity_type}')
 def step_impl(context, attribute, expected_entity_type):
 
+    expected_entity_types = tuple(map(str.strip, expected_entity_type.split(' or ')))
+
     def _():
         for inst in context.instances:
             related_entity = misc.map_state(inst, lambda i: getattr(i, attribute, None))
             errors = []
             def accumulate_errors(i):
-                if not i.is_a(expected_entity_type):
+                if not any(i.is_a().lower() == x.lower() for x in expected_entity_types):
                     misc.map_state(inst, lambda x: errors.append(err.AttributeTypeError(False, x, [i], attribute, expected_entity_type)))
```

Comment:
```suggestion
                    misc.map_state(inst, lambda x: errors.append(err.AttributeTypeError(False, x, i, attribute, expected_entity_type)))
```
Change `[i] `to` i`

It's not code that is new in this PR I see. The tests for
_'fail-als006-sectioned-solid-horizontal-wrong-item.ifc'_ 
returns the error
`expected type IfcCompositeCurve .... but found [#79=IfcGradientCurve((#8..#54,$)]`

Which gives the false impression that the error is due to the entity being in a list `['IfcGradientCurve']` rather than the  entity itself` 'IfcGradientCurve'`. 
When running the test, it appears that it does evaluate the value 'IfcGradientCurve' (in line 79, the line above), so it's just the text output. 

I'm almost sure that this usage of the list pertains to an outdated error rendering class, possibly related to one of the alignment rules. Omitting it has no impact when executing pytest.